src/main/java/com/mraof/minestuck/client/gui/SummonerScreen.java

```diff
@@ -0,0 +1,114 @@
+	
+	private static final int guiWidth = 150;
+	private static final int guiHeight = 98;
+	
+	private final SummonerTileEntity te;
+	private boolean isUntriggerable;
+	private int summonRange;
+	
+	public Button incrementButton;
+	public Button decrementButton;
+	private Button unTriggerableButton;
+	
+	private TextFieldWidget entityTypeTextField;
+	
+	
+	SummonerScreen(SummonerTileEntity te)
+	{
+		super(new StringTextComponent("Summoner"));
+		
+		this.te = te;
+		this.summonRange = te.getSummonRange() > 0 ? te.getSummonRange() : 8; //if its defaulted on creation to 0, set it to the intended default of 8
+		this.isUntriggerable = te.getBlockState().getValue(SummonerBlock.UNTRIGGERABLE);
+	}
+	
+	@Override
+	public void init()
+	{
+		int yOffset = (this.height / 2) - (guiHeight / 2);
+		
+		addButton(incrementButton = new ExtendedButton(this.width / 2 + 20, (height - guiHeight) / 2 + 12, 20, 20, new StringTextComponent("+"), button -> changeRange(1)));
+		addButton(decrementButton = new ExtendedButton(this.width / 2 - 40, (height - guiHeight) / 2 + 12, 20, 20, new StringTextComponent("-"), button -> changeRange(-1)));
+		
+		this.entityTypeTextField = new TextFieldWidget(this.font, this.width / 2 - 60, yOffset + 40, 120, 18, new StringTextComponent("Current Entity Type"));	//TODO Use translation instead, and maybe look at other text fields for what the text should be
+		this.entityTypeTextField.setValue(EntityType.getKey(te.getSummonedEntity()).toString());
+		addButton(entityTypeTextField);
+		
+		if(isUntriggerable)
+			addButton(unTriggerableButton = new ExtendedButton(this.width / 2 - 65, yOffset + 70, 85, 20, new StringTextComponent("UNTRIGGERABLE"), button -> cycleUntriggerable()));
+		else
+			addButton(unTriggerableButton = new ExtendedButton(this.width / 2 - 65, yOffset + 70, 85, 20, new StringTextComponent("TRIGGERABLE"), button -> cycleUntriggerable()));
+		addButton(new ExtendedButton(this.width / 2 + 25, yOffset + 70, 40, 20, new StringTextComponent("DONE"), button -> finish()));
+	}
+	
+	/**
+	 * Attempts to increase or decrease the range at which the entity can be summoned if that value is between the min and max
+	 */
+	private void changeRange(int change)
+	{
+		summonRange = MathHelper.clamp(summonRange + change, 1, 64);
+	}
+	
+	/**
+	 * Cycles between the block being triggerable and untriggerable
+	 */
+	private void cycleUntriggerable()
+	{
+		isUntriggerable = !isUntriggerable;
+		if(isUntriggerable)
+			unTriggerableButton.setMessage(new StringTextComponent("UNTRIGGERABLE"));
+		else
+			unTriggerableButton.setMessage(new StringTextComponent("TRIGGERABLE"));
+	}
+	
+	/**
+	 * Returns the current entity type, with imps as the generic entity type
+	 */
+	private EntityType<?> getEntityType(String stringInput)
+	{
+		Optional<EntityType<?>> attemptedEntityType = EntityType.byString(stringInput);
+		return attemptedEntityType.orElse(MSEntityTypes.IMP);
```

Comment:
It may make sense to change the gui so that it rejects nonexisting entity types (draws a message on the gui and doesn't allow the user to press the "DONE" button until a valid type is written), instead of silently changing it to an imp.

Reply:
Did this, but instead of a message I turned the text in the text field red and prevented it from sending the packet/exiting